gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/CodeableConceptMapper.java

```diff
@@ -40,22 +177,80 @@ private CD getSnomedTranslationElement(CD codedData) {
             .get();
     }
 
-    private String getText(CD codedData) {
-        return codedData.getOriginalText() != null ? codedData.getOriginalText() : codedData.getDisplayName();
+    private String generateText(String text, String newText) {
+        return StringUtils.isNotEmpty(text) ? text : newText;
     }
 
     private String getDisplayForSnomedTranslationElement(CD codedData, CD translation) {
         return translation.getDisplayName() != null ? translation.getDisplayName() : codedData.getDisplayName();
     }
 
-    private CodeableConcept createCodeableConcept(String code, String system, String display, String text) {
+    private boolean isCodeConceptId(String partitionIdentifier) {
+        return CONCEPT_PARTITION_SHORT.equals(partitionIdentifier) || CONCEPT_PARTITION_LONG.equals(partitionIdentifier);
+    }
+
+    private boolean isCodeDescriptionId(String partitionIdentifier) {
+        return DESCRIPTION_PARTITION_SHORT.equals(partitionIdentifier) || DESCRIPTION_PARTITION_LONG.equals(partitionIdentifier);
+    }
+
+    private String getPartitionIdentifier(String code) {
+        var codeLength = code.length();
+        return code.substring(codeLength - MAX_CHARACTERS_FROM_RIGHT, codeLength - MIN_CHARACTERS_FROM_RIGHT);
+    }
+
+    private CodeableConcept createCodeableConcept(String code, String system, String display, String text, Extension extension) {
         var codeableConcept = new CodeableConcept();
         codeableConcept
             .setText(text)
             .getCodingFirstRep()
             .setCode(code)
             .setSystem(system)
-            .setDisplay(display);
+            .setDisplay(display)
+            .addExtension(extension);
         return codeableConcept;
     }
+
+    private SnomedCTDescription getSnomedDescriptionUsingConceptIdDisplayName(String conceptId, String displayName) {
```

Comment:
I think that private methods that do nothing more than invoke another method on snomedCTDao are not necessary